cli/medperf/commands/compatibility_test/run.py

```diff
@@ -0,0 +1,243 @@
+
+
+        Args:
+            benchmark (int, optional): Benchmark to run the test workflow for
+            data_prep (str, optional): data preparation mlcube uid or local path.
+            model (str, optional): model mlcube uid or local path.
+            evaluator (str, optional): evaluator mlcube uid or local path.
+            data_path (str, optional): path to a local raw data
+            labels_path (str, optional): path to the labels of the local raw data
+            demo_dataset_url (str, optional): Identifier to download the demonstration dataset tarball file.\n
+            See `medperf mlcube submit --help` for more information
+            demo_dataset_hash (str, optional): The hash of the demo dataset tarball file
+            data_uid (str, optional): A prepared dataset UID
+            no_cache (bool): Whether to ignore cached results of the test execution. Defaults to False.
+            offline (bool): Whether to disable communication to the MedPerf server and rely only on
+            local copies of the server assets. Defaults to False.
+
+        Returns:
+            (str): Prepared Dataset UID used for the test. Could be the one provided or a generated one.
+            (dict): Results generated by the test.
+        """
+        logging.info("Starting test execution")
+        test_exec = cls(
+            benchmark,
+            data_prep,
+            model,
+            evaluator,
+            data_path,
+            labels_path,
+            demo_dataset_url,
+            demo_dataset_hash,
+            data_uid,
+            no_cache,
+            offline,
+        )
+        test_exec.validate()
+        test_exec.process_benchmark()
+        test_exec.prepare_cubes()
+        test_exec.prepare_dataset()
+        test_exec.initialize_report()
+        results = test_exec.cached_results()
+        if results is None:
+            results = test_exec.execute()
+            test_exec.write(results)
+        return test_exec.data_uid, results
+
+    def __init__(
+        self,
+        benchmark: int = None,
+        data_prep: str = None,
+        model: str = None,
+        evaluator: str = None,
+        data_path: str = None,
+        labels_path: str = None,
+        demo_dataset_url: str = None,
+        demo_dataset_hash: str = None,
+        data_uid: str = None,
+        no_cache: bool = False,
+        offline: bool = False,
+    ):
+        self.benchmark_uid = benchmark
+        self.data_prep = data_prep
+        self.model = model
+        self.evaluator = evaluator
+        self.data_path = data_path
+        self.labels_path = labels_path
+        self.demo_dataset_url = demo_dataset_url
+        self.demo_dataset_hash = demo_dataset_hash
+        self.data_uid = data_uid
+        self.no_cache = no_cache
+        self.offline = offline
+
+        # This property will be set by the validator class
+        # to either "path", "demo", "prepared", or "benchmark"
+        self.data_source = None
```

Comment:
This was also confusing. It is not known from reading this file at which time and how this property is being assigned. It would be better to have this call something like `self.data_source = self.validator.get_data_source()`